### PR TITLE
Feature/bluetooth

### LINKED ISSUE #16: [Feature Request]: Bluetooth communication
# Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

# Expected Behavior

The Robot makes a wireless connection with a pc and can receive commands 

# Current Behavior

The robot cannot make a wireless connection via Bluetooth and cannot receive commands

### PR DESCRIPTION
**Pull request by Daantuk01**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #16

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please provide the name of the unit test that has been created/modified. 
Please also include the steps that the test takes to verify the test is reliable and works.

**Test A**
1. Make a connection via Putty
2. Press the wasd keys slowly to see if the robot will move

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/RobotMain.java [hunk 1]
```diff
@@ -45,7 +44,7 @@ private void initialize() {
 
         this.movement = new Movement(this);
         this.processes.add(this.movement);
-        this.movement.forward();
+//        this.movement.forward();
```
Comment (Daantuk01) [inline src/RobotMain.java:47]: delete comment

File: src/RobotMain.java [hunk 2]
```diff
@@ -55,6 +54,10 @@ private void initialize() {
 
         this.blinkers = new Blinkers();
         this.processes.add(this.blinkers);
+
+        this.wirelessConnection = new WirelessConnection(this);
+        this.processes.add(this.wirelessConnection);
+
```
Comment (Daantuk01) [inline src/RobotMain.java:60]: delete whitespace

File: src/RobotMain.java [hunk 3]
```diff
@@ -117,4 +120,24 @@ public void onCollisionDetectionUpdate(WhiskerStatus whiskerCollision) {
...
+        } else if (data == WirelessConfig.left) {
+            this.movement.turnLeft();
+        } else if (data == WirelessConfig.right) {
+            this.movement.turnRight();
+        } else if (data == WirelessConfig.forward) {
+            this.movement.forward();
+        } else if (data == WirelessConfig.stop) {
+            this.movement.neutral();
+        } else if (data == WirelessConfig.transfer) {
+
```
Comment (Daantuk01) [inline src/RobotMain.java:140]: delete whiteline

File: src/vehicle/WirelessConnection.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    /**
+     * Checks when wireless conection is available and calls the callback and gives the data
+     */
+    public void update() {
+        if (this.bluetooth.available() > 0) {
+            int data = bluetooth.readByte();
+            bluetooth.writeByte(data); // Echo data
+            callback.onWirelessUpdate(data);
+            System.out.println("Received: " + data);
```
Comment (Daantuk01) [inline src/vehicle/WirelessConnection.java:29]: delete system outprint

**Review by KaanMol — APPROVED** [review]

**Pull request closed without merging**